core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java

```diff
@@ -96,7 +96,7 @@
             if (instance instanceof AbstractInfinispanTest) {
                String instanceName = ((AbstractInfinispanTest) instance).getTestName();
                Object otherInstance = instancesByName.putIfAbsent(instanceName, instance);
-               if (otherInstance != null) {
+               if (otherInstance != null && instance.getClass() != otherInstance.getClass()) {
```

Comment:
The interceptor only processes the methods of one class (i.e. TestNG XML test) at once, so this condition will never be true. Better to keep the condition but log something to the console instead of throwing the exception and skipping all the methods.

Reply:
Done!